### PR TITLE
Sys 3734 validators manager compat eth bridge

### PR DESCRIPTION
**Pull request by ivan-cholakov**

### CODE REVIEW

File: pallets/validators-manager/src/lib.rs [hunk 1]
```diff
@@ -800,23 +473,25 @@ impl<T: Config> Pallet<T> {
...
+            .map_err(|_| Error::<T>::InvalidPublicKey)?;
+
 
-        let tx_id = 0;
+        let function_name = b"removeAuthor";
+        let params = vec![            
+            (b"bytes32".to_vec(), <T as pallet::Config>::AccountToBytesConvert::into_bytes(validator_id).to_vec()),
+            (b"bytes".to_vec(), decompressed_eth_public_key.to_fixed_bytes().to_vec()),
+        ];
+        let tx_id = <T as pallet::Config>::BridgePublisher::publish(function_name, &params).map_err(|e| DispatchError::Other(e.into()))?;
```
Comment (fluorostani) [inline pallets/validators-manager/src/lib.rs:485]: Maybe pull out validator_id_bytes

File: pallets/validators-manager/src/lib.rs [hunk 2]
```diff
@@ -800,23 +473,25 @@ impl<T: Config> Pallet<T> {
...
-            AVN::<T>::calculate_primary_validator(<system::Pallet<T>>::block_number())
-                .map_err(|_| Error::<T>::ErrorCalculatingPrimaryValidator)?;
+        let decompressed_eth_public_key = decompress_eth_public_key(t1_eth_public_key)
+            .map_err(|_| Error::<T>::InvalidPublicKey)?;
+
 
-        let tx_id = 0;
+        let function_name = b"removeAuthor";
+        let params = vec![            
+            (b"bytes32".to_vec(), <T as pallet::Config>::AccountToBytesConvert::into_bytes(validator_id).to_vec()),
```
Comment (fluorostani) [inline pallets/validators-manager/src/lib.rs:482]: Maybe pull out validator_id_bytes

File: pallets/validators-manager/src/lib.rs [hunk 3]
```diff
@@ -773,7 +446,7 @@ impl<T: Config> Pallet<T> {
         validator_id: &T::AccountId,
         ingress_counter: IngressCounter,
         action_type: ValidatorsActionType,
-        eth_transaction_type: EthTransactionType,
+        t1_eth_public_key: ecdsa::Public
```
Comment (fluorostani) [inline pallets/validators-manager/src/lib.rs:449]: I would say use either "t1_public_key" or "eth_public_key"

File: pallets/validators-manager/src/lib.rs [hunk 4]
```diff
@@ -725,28 +393,33 @@ impl<T: Config> Pallet<T> {
...
-        let (new_validator_id, eth_tx_sender, eth_transaction_type, tx_id) =
-            Self::prepare_registration_data(collator_eth_public_key, collator_account_id)?;
-
+        let decompressed_eth_public_key =
+            decompress_eth_public_key(*collator_eth_public_key)
+                .map_err(|_| Error::<T>::InvalidPublicKey)?;
+        let function_name = b"addAuthor";
+        let params = vec![            
+            (b"bytes".to_vec(), decompressed_eth_public_key.to_fixed_bytes().to_vec()),
+            (b"bytes32".to_vec(), <T as pallet::Config>::AccountToBytesConvert::into_bytes(collator_account_id).to_vec()),
```
Comment (fluorostani) [inline pallets/validators-manager/src/lib.rs:412]: Maybe pull out validator_id_bytes

**Review by fluorostani — APPROVED** [review]

**Review by nahuseyoum — APPROVED** [review]

File: pallets/validators-manager/src/lib.rs [hunk 5]
```diff
@@ -667,6 +678,22 @@ impl<T: Config> NewSessionHandler<T::AuthorityId, T::AccountId> for Pallet<T> {
                         ingress_counter,
                         
                     );
+                } else if validators_action_data.status == ValidatorsActionStatus::AwaitingConfirmation &&
+                validators_action_data.action_type.is_activation() {
```
Comment (nahuseyoum) [inline pallets/validators-manager/src/lib.rs:682]: Why do we need this branch here? Was it not working previously?

Comment (ivan-cholakov) [reply pallets/validators-manager/src/lib.rs:682]: this branch existed before it initiated the activation 
```
else if Self::can_setup_voting_to_activate_validator(
                    &validators_action_data,
                    &action_account_id,
                    active_validators,
                ) {
                    Self::setup_voting_to_activate_validator(
                        ingress_counter,
                        &action_account_id,
                        calculate_two_third_quorum(active_validators.len() as u32),
                        voting_period_end,
                    );
                }
```

so now we just activate the validator

**Pull request merged**